pkg/logql/evaluator.go

```diff
@@ -123,31 +123,31 @@ func Sortable(q Params) (bool, error) {
 	return sortable, nil
 }
 
-// Evaluator is an interface for iterating over data at different nodes in the AST
-type Evaluator interface {
-	SampleEvaluator
-	EntryEvaluator
+// EvaluatorFactory is an interface for iterating over data at different nodes in the AST
+type EvaluatorFactory interface {
+	SampleEvaluatorFactory
+	EntryEvaluatorFactory
 }
```

Comment:
`Factory` feels really java'esque. Happy to change the name.

Reply:
I think the more idiomatic Go name is _Builder_, but clarity is what we need here and this is clear so 👍